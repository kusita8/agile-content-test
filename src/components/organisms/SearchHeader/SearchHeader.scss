.searchheader {
  @media only screen and (min-width: 600px) {
    width: 70rem;
    display: flex;
    gap: 3rem;
    align-items: center;
  }

  &__search-container {
    @media only screen and (min-width: 600px) {
      width: 45rem;
    }
  }

  &__logo {
    width: 9.2rem;
    display: block;

    img {
      width: 100%;
    }
  }
}
