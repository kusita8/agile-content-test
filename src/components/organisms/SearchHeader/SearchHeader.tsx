import { useCallback } from "react";
import { Link } from "react-router-dom";

import { Search } from "src/components/molecules/Search/Search";
import { useSearchQueryParams } from "src/query/useSearchQueryParams";
import { Path } from "src/routes/Path";

import "./SearchHeader.scss";

import googleLogo from "/src/assets/google-logo.png";

export const SearchHeader = () => {
  const setParam = useSearchQueryParams((state) => state.setParam);
  const searchQuery = useSearchQueryParams((state) => state.params.query);

  const handleSubmit = useCallback((value: string) => {
    setParam("query", value);
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  return (
    <div className="searchheader">
      <Link className="searchheader__logo" to={Path.HOME}>
        <img src={googleLogo} alt="Google logo" />
      </Link>
      <div className="searchheader__search-container">
        <Search
          key={searchQuery}
          defaultValue={searchQuery}
          size="small"
          onSubmit={handleSubmit}
        />
      </div>
    </div>
  );
};
