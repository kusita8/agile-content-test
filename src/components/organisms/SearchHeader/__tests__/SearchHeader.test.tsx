import { SearchHeader } from "src/components/organisms/SearchHeader/SearchHeader";
import { renderWithProviders } from "src/tests/renderWithProviders";

describe("SearchHeader", () => {  
  test("it should match the snapshot", () => {
    const { asFragment } = renderWithProviders(<SearchHeader />);
    expect(asFragment()).toMatchSnapshot();
  });
});
