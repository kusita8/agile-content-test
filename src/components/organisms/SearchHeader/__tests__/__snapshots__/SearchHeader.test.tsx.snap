// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SearchHeader > it should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="layout"
  >
    <div
      class="header"
    >
      <p>
        <b>
          Agile content
        </b>
         Frontend test
      </p>
      <div
        class="header__right"
      >
        <svg
          class="gb_p"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M6,8c1.1,0 2,-0.9 2,-2s-0.9,-2 -2,-2 -2,0.9 -2,2 0.9,2 2,2zM12,20c1.1,0 2,-0.9 2,-2s-0.9,-2 -2,-2 -2,0.9 -2,2 0.9,2 2,2zM6,20c1.1,0 2,-0.9 2,-2s-0.9,-2 -2,-2 -2,0.9 -2,2 0.9,2 2,2zM6,14c1.1,0 2,-0.9 2,-2s-0.9,-2 -2,-2 -2,0.9 -2,2 0.9,2 2,2zM12,14c1.1,0 2,-0.9 2,-2s-0.9,-2 -2,-2 -2,0.9 -2,2 0.9,2 2,2zM16,6c0,1.1 0.9,2 2,2s2,-0.9 2,-2 -0.9,-2 -2,-2 -2,0.9 -2,2zM12,8c1.1,0 2,-0.9 2,-2s-0.9,-2 -2,-2 -2,0.9 -2,2 0.9,2 2,2zM18,14c1.1,0 2,-0.9 2,-2s-0.9,-2 -2,-2 -2,0.9 -2,2 0.9,2 2,2zM18,20c1.1,0 2,-0.9 2,-2s-0.9,-2 -2,-2 -2,0.9 -2,2 0.9,2 2,2z"
          />
        </svg>
        <div
          class="header__user"
        >
          <img
            alt="Profile image"
            src="https://images.unsplash.com/photo-1511367461989-f85a21fda167?ixlib=rb-4.0.3&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=2231&q=80"
          />
        </div>
      </div>
    </div>
    <div
      class="searchheader"
    >
      <a
        class="searchheader__logo"
        href="/"
      >
        <img
          alt="Google logo"
          src="/src/assets/google-logo.png"
        />
      </a>
      <div
        class="searchheader__search-container"
      >
        <div
          class="search search--small"
        >
          <form
            id="search-form"
          >
            <svg
              focusable="false"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M15.5 14h-.79l-.28-.27A6.471 6.471 0 0 0 16 9.5 6.5 6.5 0 1 0 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
              />
            </svg>
            <input
              data-testid="search-input"
              placeholder=" "
              type="text"
              value=""
            />
            <button
              type="button"
            >
              <svg
                focusable="false"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                />
              </svg>
            </button>
          </form>
        </div>
      </div>
    </div>
    <div
      class="footer"
    >
      <p>
        Â© Google 2023
      </p>
      <p>
        version: 0.0.0
      </p>
    </div>
  </div>
</DocumentFragment>
`;
